cql3/statements/select_statement.cc

```diff
@@ -1651,10 +1651,16 @@ void select_statement::check_needs_filtering(const restrictions::statement_restr
 {
     // non-key-range non-indexed queries cannot involve filtering underneath
     if (!_parameters->allow_filtering() && (restrictions.is_key_range() || restrictions.uses_secondary_indexing())) {
+        const auto& ck_restrictions = *restrictions.get_clustering_columns_restrictions();
```

Comment:
Note on the commit message (github doesn't let me add comments on that...) - it's not just "multiple partitions" - a discrete number of partitions would have been fine. The problem is a request (scan of the full table or as little as a single token), which may result in an **unlimited number** of partitions.

Reply:
Done.